tests/model_tests/test_protocols.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+from nsot import exc, models
+from .fixtures import circuit, site
+
+
+# Allow everything in there to access the DB
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture
+def asn_attribute(site):
+    return models.Attribute.objects.create(
+        site=site,
+        name='asn',
+        resource_name='Protocol',
+    )
+
+
+@pytest.fixture
+def bgp(asn_attribute):
+    bgp = models.ProtocolType.objects.create(
+        name='bgp',
+        description='Border Gateway Protocol',
+    )
+    bgp.required_attributes.add(asn_attribute)
```

Comment:
One thing I haven't seen is how we might create a new `ProtocolType` instance w/ the `required_attributes` specified up front. We need to consider this use-case because how user interfaces might interact with these objects.

Reply:
Is that even possible with a `ManyToManyField`? The Django docs indicate that the `ProtocolType` object would have to be saved before any associations with attributes (via `required_attributes`) can be made.

Find in page "You can’t associate it with a Publication until it’s been saved":
https://docs.djangoproject.com/en/1.8/topics/db/examples/many_to_many/